lib/ansible/module_utils/urls.py

```diff
@@ -1049,11 +1049,36 @@ def fetch_url(module, url, data=None, headers=None, method=None,
                      url_password=password, http_agent=http_agent, force_basic_auth=force_basic_auth,
                      follow_redirects=follow_redirects, client_cert=client_cert,
                      client_key=client_key, cookies=cookies)
-        info.update(r.info())
+        # Lowercase keys, to conform to py2 behavior, so that py3 and py2 are predictable
+        info.update(dict((k.lower(), v) for k, v in r.info().items()))
+
+        # Don't be lossy, append header values for duplciate headers
+        try:
+            # Py3
+            _h = {}
+            for k, v in r.headers.items():
+                # The same as above, lower case keys to match py2 behavior, and create more consistent results
+                _k = k.lower()
+                if _k in _h:
+                    _h[_k] = ', '.join((_h[_k], v))
+                else:
+                    _h[_k] = v
+            info.update(_h)
```

Comment:
Please use more descriptive variable names for at least `_h` and `_k`, and probably `k` and `v` also. I think it's fine to use some single letter variables in list/dict comprehensions or simple loops, but once there is some additional logic I think readability is improved by using longer names.

Also, it should be fine to reassign the lowercase version of `k` (or whatever it ends up being) to itself since the original value isn't used anywhere else and the type hasn't changed.